src/main/java/org/variantsync/diffdetective/diff/difftree/DiffType.java

```diff
@@ -53,12 +55,18 @@ public void matchBeforeAfter(final Runnable ifExistsBefore, final Runnable ifExi
      * Runs task on ifExistsBefore if the value existed before the edit (DiffType != ADD).
      * Runs task on ifExistsAfter if the value exists after the edit (DiffType != ADD).
      * Note: Runs task on both arguments sequentially if the artifact was not edited (DiffType == NON).
+     * Some tasks may not be run if a task throws an exception.
      *
      * @param ifExistsBefore Argument that is valid if the diff did not add.
      * @param ifExistsAfter Argument that is valid if the edit did not remove.
      * @param task Task to run with all given arguments that are valid w.r.t. to this DiffType's lifetime.
+     * @throws E iff {@code task} throws {@code E}
      */
-    public <T> void matchBeforeAfter(final T ifExistsBefore, final T ifExistsAfter, final Consumer<T> task) {
+    public <T, E extends Throwable> void matchBeforeAfter(
+        final T ifExistsBefore,
+        final T ifExistsAfter,
+        final FailableConsumer<T, E> task
+    ) throws E {
```

Comment:
This cannot be called anymore with a default consumer, right? Would be nice to be able to run this method with a non-failing consumer but as long as we do not need it right now, we do not have to add it.

Reply:
It cannot be called if you have an explicit instance of `Consumer`, for example using the old anonymous class style: 
```java
diffType.matchBeforeAfter(1, 2, new Consumer() {
  @Override
  void accept(Integer a) {
    // Do something with a
  }
}
```
If you instead call it with a lambda it works just fine (if the lambda does not throw any exceptions the type `E` is resolved to `RuntimeException` and `FailableConsumer<T, RuntimeException>` and `Consumer<T>` are functionally equivalent).

However, when adding an overload for `Consumer<T>` the overload resolutions fails when using a lambda, which would be rather unfortunate.